macros/plugins/spark/helpers/recover_partitions.sql

```diff
@@ -0,0 +1,12 @@
+{% macro spark__recover_partitions(source_node) %}
```

Comment:
I kept this Spark specific. If we want to have the same flow a for the other macros, we need to a `recover_partitions` macro to common and use the dispatch to come back to the spark one